demo-apps/cf-plugtest-client/src/main/java/org/eclipse/californium/plugtests/PlugtestClient.java

```diff
@@ -705,6 +719,103 @@ public static void testCL(String uri, EndpointContext context) throws ConnectorE
+	 * @throws ConnectorException on connection failure
+	 * @throws IOException on IO failure
+	 */
+	public static void testOscore(String uri, EndpointContext context) throws ConnectorException, IOException {
+
+		// re-usable response object
+		CoapResponse response;
+
+		CoapClient client = new CoapClient(uri + "/oscore");
+		client.setDestinationContext(context);
+
+		System.out.println("\nWill run OSCORE Appendix B.2 context rederivaton followed by 4 OSCORE requests.");
+
+		System.out.println("===============\nOSCORE 01");
+		System.out.println("---------------\nGET /test\n---------------");
+		Request r = Request.newGet();
+		r.getOptions().setOscore(Bytes.EMPTY);
+		response = client.advanced(r);
+		context = client.getDestinationContext();
+		if (context != null) {
+			System.out.println(Utils.prettyPrint(context));
+		}
+		System.out.println(response.advanced().getType() + "-" + response.getCode());
+		System.out.println(response.getResponseText());
+
+		System.out.println("===============\nOSCORE 02");
+		System.out.println("---------------\nDELETE /test\n---------------");
+		r = Request.newDelete();
+		r.getOptions().setOscore(Bytes.EMPTY);
+		response = client.advanced(r);
+		System.out.println(response.advanced().getType() + "-" + response.getCode());
+		System.out.println(response.getResponseText());
+
+		System.out.println("===============\nOSCORE 03");
+		System.out.println("---------------\nPUT /test\n---------------");
+		r = Request.newPut();
+		r.setPayload("");
+		r.getOptions().setContentFormat(MediaTypeRegistry.TEXT_PLAIN);
+		r.getOptions().setOscore(Bytes.EMPTY);
+		response = client.advanced(r);
+		System.out.println(response.advanced().getType() + "-" + response.getCode());
+		System.out.println(response.getResponseText());
+
+		System.out.println("===============\nOSCORE 04");
+		System.out.println("---------------\nPOST /test\n---------------");
+		r = Request.newPut();
+		r.setPayload("non-empty");
+		r.getOptions().setContentFormat(MediaTypeRegistry.TEXT_PLAIN);
+		r.getOptions().setOscore(Bytes.EMPTY);
+		response = client.advanced(r);
+		System.out.println(response.advanced().getType() + "-" + response.getCode());
+		System.out.println(response.getResponseText());
+		
+	}
+
+	/**
+	 * Initializes an OSCORE context for the client built on a pre-defined
+	 * configuration and adds it to the OSCORE context database. The created
+	 * context will support and by default always use the Appendix B.2 context
+	 * rederivation procedure. This is to avoid message replay errors that would
+	 * otherwise arise for clients.
+	 * 
+	 * @param uri the URI of the OSCORE resource at the server
+	 * @param db the OSCORE context database
+	 */
+	public static void initOscore(String uri, HashMapCtxDB db) {
+		AlgorithmID alg = AlgorithmID.AES_CCM_16_64_128;
+		AlgorithmID kdf = AlgorithmID.HKDF_HMAC_SHA_256;
+
+		byte[] master_secret = { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E,
```

Comment:
Maybe you have a look at [StringUtil.hex2ByteArray(String)](https://github.com/eclipse/californium/blob/master/element-connector/src/main/java/org/eclipse/californium/elements/util/StringUtil.java#L248). At least for initialization, that may be easier.

Reply:
Done in commit e0ea3913838eecdd4908b73034ba5aa8a41858de